### PR TITLE
feat(context): migrate base connector [STEP 2]

### PR DESCRIPTION
**Pull request by Haroenv**
This is enough changes to be able to look at the storybook without errors again (only pagination, hits, searchbox of course, and nothing multi-index)

For full roadmap, see #2178

[→ stories](
https://deploy-preview-2179--react-instantsearch.netlify.com/storybook/?selectedKind=Configure&selectedStory=default&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs)

### CODE REVIEW

File: packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js [hunk 1]
```diff
@@ -5,39 +5,61 @@ jest.mock('../../core/createConnector', () => x => x);
...
-    const context = { context: { ais: { mainTargetedIndex: 'index' } } };
-    const getSearchParameters = connect.getSearchParameters.bind(context);
-    const transitionState = connect.transitionState.bind(context);
-    const cleanUp = connect.cleanUp.bind(context);
+    const contextValue = { ais: { mainTargetedIndex: 'index' } };
 
-    it('propagates the props to the SearchParameters without children', () => {
-      const searchParameters = getSearchParameters(
+    it('propagates the props to the SearchParameters without children & contextValue', () => {
+      const searchParameters = connect.getSearchParameters(
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js [hunk 2]
```diff
@@ -5,39 +5,61 @@ jest.mock('../../core/createConnector', () => x => x);
...
+        `"Parameter 'contextValue' is not an attribute of SearchParameters (http://algolia.github.io/algoliasearch-helper-js/docs/SearchParameters.html)"`
+      );
     });
 
     it('calling transitionState should add configure parameters to the search state', () => {
-      const providedThis = {};
-      let searchState = transitionState.call(
-        providedThis,
-        { distinct: 1, whatever: 'please', children: 'whatever' },
+      let searchState = connect.transitionState(
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js [hunk 3]
```diff
@@ -46,22 +68,32 @@ describe('connectConfigure', () => {
     });
 
     it('calling cleanUp should remove configure parameters from the search state', () => {
-      let searchState = cleanUp(
-        { distinct: 1, whatever: 'please', children: 'whatever' },
+      let searchState = connect.cleanUp(
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectHits.js [hunk 4]
```diff
@@ -6,18 +6,20 @@ const { getSearchParameters } = connect;
...
 describe('connectHits', () => {
   describe('single index', () => {
-    const context = { context: { ais: { mainTargetedIndex: 'index' } } };
-    const getProvidedProps = connect.getProvidedProps.bind(context);
+    const contextValue = { mainTargetedIndex: 'index' };
+
     it('provides the current hits to the component', () => {
       const hits = [{}];
-      const props = getProvidedProps(null, null, {
+      const props = connect.getProvidedProps({ contextValue }, null, {
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectHits.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectPagination.js [hunk 5]
```diff
@@ -8,23 +8,22 @@ let params;
...
-    const getProvidedProps = connect.getProvidedProps.bind(context);
-    const refine = connect.refine.bind(context);
-    const getSP = connect.getSearchParameters.bind(context);
-    const getMetadata = connect.getMetadata.bind(context);
-    const cleanUp = connect.cleanUp.bind(context);
+    const contextValue = { mainTargetedIndex: 'index' };
 
     it('provides the correct props to the component', () => {
-      props = getProvidedProps({}, {}, { results: { nbPages: 666, hits: [] } });
+      props = connect.getProvidedProps(
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectPagination.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js [hunk 6]
```diff
@@ -8,21 +8,22 @@ let params;
...
-    const context = { context: { ais: { mainTargetedIndex: 'index' } } };
-    const getProvidedProps = connect.getProvidedProps.bind(context);
-    const refine = connect.refine.bind(context);
-    const getSP = connect.getSearchParameters.bind(context);
-    const cleanUp = connect.cleanUp.bind(context);
+    const contextValue = { mainTargetedIndex: 'index' };
+
     it('provides the correct props to the component', () => {
-      props = getProvidedProps({}, {}, {});
+      props = connect.getProvidedProps({ contextValue }, {}, {});
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js [hunk 7]
```diff
@@ -8,21 +8,22 @@ let params;
...
       expect(props).toEqual({ currentRefinement: '' });
 
-      props = getProvidedProps({}, { query: 'yep' }, {});
+      props = connect.getProvidedProps({ contextValue }, { query: 'yep' }, {});
       expect(props).toEqual({ currentRefinement: 'yep' });
     });
 
     it("calling refine updates the widget's search state", () => {
-      const nextState = refine({}, { otherKey: 'val' }, 'yep');
+      const nextState = connect.refine(
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js [hunk 8]
```diff
@@ -31,19 +32,29 @@ describe('connectSearchBox', () => {
...
+          {}
+        )
+      ).toEqual({
         currentRefinement: 'yaw',
       });
     });
 
     it('refines the query parameter', () => {
-      params = getSP(new SearchParameters(), {}, { query: 'bar' });
+      params = connect.getSearchParameters(
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js [hunk 9]
```diff
@@ -31,19 +32,29 @@ describe('connectSearchBox', () => {
...
+        { contextValue },
+        { query: 'bar' }
+      );
       expect(params.query).toBe('bar');
     });
 
     it('should return the right searchState when clean up', () => {
-      const searchState = cleanUp(
-        {},
+      const searchState = connect.cleanUp(
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectSearchBox.js]: We should keep the test bound to the context, we never know.

File: packages/react-instantsearch-core/src/connectors/connectConfigure.js [hunk 10]
```diff
@@ -16,37 +16,42 @@ export default createConnector({
     return {};
   },
   getSearchParameters(searchParameters, props) {
-    const items = omit(props, 'children');
+    const items = omit(props, 'children', 'contextValue');
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/connectConfigure.js]: We can avoid omit with destructuring:

```
const { children, contextValue, ...items } = props;
```

File: packages/react-instantsearch-core/src/connectors/connectConfigure.js [hunk 11]
```diff
@@ -16,37 +16,42 @@ export default createConnector({
...
     const nonPresentKeys = this._props
-      ? difference(keys(this._props), keys(props))
+      ? difference(Object.keys(this._props), Object.keys(props))
       : [];
     this._props = props;
     const nextValue = {
       [id]: { ...omit(nextSearchState[id], nonPresentKeys), ...items },
     };
-    return refineValue(nextSearchState, nextValue, this.context);
+    return refineValue(nextSearchState, nextValue, { ais: props.contextValue });
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/connectConfigure.js]: At the moment this is not correct, we should be very mindful when we review the PR on the multi-index.

File: packages/react-instantsearch-core/src/connectors/connectConfigure.js [hunk 12]
```diff
@@ -16,37 +16,42 @@ export default createConnector({
...
   },
   cleanUp(props, searchState) {
     const id = getId();
-    const indexId = getIndexId(this.context);
+    const indexId = getIndexId({ ais: props.contextValue });
+
     const subState =
-      hasMultipleIndices(this.context) && searchState.indices
+      hasMultipleIndices(props.contextValue) && searchState.indices
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/connectConfigure.js]: Once we complete the migration we should create a function that extract the values from props:

```
const { state, target } = getContext(props);
```

File: packages/react-instantsearch-core/src/connectors/connectConfigure.js (hunk 10 above)
Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/connectConfigure.js]: going to change that in #2185

File: packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js [hunk 13]
```diff
@@ -5,39 +5,65 @@ jest.mock('../../core/createConnector', () => x => x);
 
 describe('connectConfigure', () => {
   describe('single index', () => {
-    const context = { context: { ais: { mainTargetedIndex: 'index' } } };
-    const getSearchParameters = connect.getSearchParameters.bind(context);
-    const transitionState = connect.transitionState.bind(context);
-    const cleanUp = connect.cleanUp.bind(context);
+    const contextValue = { ais: { mainTargetedIndex: 'index' } };
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js]: The `contextValue` seems to not be correct in that case, it should be without `ais`.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js [hunk 14]
```diff
@@ -5,39 +5,65 @@ jest.mock('../../core/createConnector', () => x => x);
...
+        `"Parameter 'contextValue' is not an attribute of SearchParameters (http://algolia.github.io/algoliasearch-helper-js/docs/SearchParameters.html)"`
+      );
     });
 
     it('calling transitionState should add configure parameters to the search state', () => {
-      const providedThis = {};
-      let searchState = transitionState.call(
-        providedThis,
-        { distinct: 1, whatever: 'please', children: 'whatever' },
+      const ctx = {};
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectConfigure.js]:
```suggestion
      const instance = {};
```

**Pull request merged**